### PR TITLE
Return proper status when stream fails

### PR DESCRIPTION
**Pull request by michalkulakowski**
### 🛠 Summary

https://jira.devtools.intel.com/browse/CVS-148629

### 🧪 Checklist

- [ ] Unit tests added.
- [ ] The documentation updated.
- [ ] Change follows security best practices.
``

### CODE REVIEW

File: external/partial.patch [hunk 1]
```diff
@@ -69,33 +69,39 @@ index c8d0501b..3c0a3df6 100644
...
-   NET_LOG(FATAL, "PartialReplyWithStatus not implemented.");
+@@ -338,12 +371,18 @@ void EvHTTPRequest::AppendResponseHeader(absl::string_view header,
+   }
  }
  
--void EvHTTPRequest::PartialReply() {
+-void EvHTTPRequest::PartialReplyWithStatus(HTTPStatusCode status) {
 -  NET_LOG(FATAL, "PartialReplyWithStatus not implemented.");
-+void EvHTTPRequest::PartialReply(std::string data) {
++void EvHTTPRequest::PartialReplyWithStatus(std::string data, int code) {
```
Comment (dkalinowski) [inline external/partial.patch:78]: why not using HTTPStatusCode but int?

File: src/http_frontend/http_graph_executor_impl.cpp [hunk 2]
```diff
@@ -117,6 +117,14 @@ Status sendErrorImpl(
     return StatusCode::OK;
 }
 
+Status sendErrorImplWithCode(
```
Comment (dkalinowski) [inline src/http_frontend/http_graph_executor_impl.cpp:120]:
```suggestion
Status sendErrorWithCodeImpl(
```

File: src/http_frontend/http_graph_executor_impl.cpp [hunk 3]
```diff
@@ -117,6 +117,14 @@ Status sendErrorImpl(
...
+    const std::string& message,
+    const int code,
```
Comment (dkalinowski) [inline src/http_frontend/http_graph_executor_impl.cpp:122]: unused param, why int?

File: src/http_rest_api_handler.cpp [hunk 4]
```diff
@@ -499,7 +499,7 @@ Status HttpRestApiHandler::processV3(const std::string_view uri, const HttpReque
         serverReaderWriter->OverwriteResponseHeader("Connection", "keep-alive");
         auto status = executor->inferStream(request, *serverReaderWriter);
         if (!status.ok()) {
-            sendErrorImpl(status.string(), *serverReaderWriter);
+            sendErrorImplWithCode(status.string(), 400, *serverReaderWriter);
```
Comment (dkalinowski) [inline src/http_rest_api_handler.cpp:502]: I dont think we need to put sendErrorImplWithCode in http_graph_executor_impl.hpp. This file is supposed to contain implementations for templated methods used in graph executor (http and grpc). I would move the implementation here

Comment (dkalinowski) [reply src/http_rest_api_handler.cpp:502]: + use enum instead of 400

Comment (dkalinowski) [inline src/http_rest_api_handler.cpp:502]: There are multiple places where we send an error using `sendErrorImpl` - why do we only return with status here? Please check if we need to also return with status in all other cases (check all usages of sendErrorImpl)

**Review by rasapala — APPROVED** [review]

**Review by dkalinowski — APPROVED** [review]

**Pull request merged**